### PR TITLE
send block instead of bytes.Buffer from chain synchronizer

### LINKED ISSUE #308: Chain synchronizer should pass a block instead of a bytes.Buffer to the chain
This caused some crashes observed on testnet.

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #308

### CODE REVIEW

**Comment by kieranhall** [comment]
# [Codecov](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=h1) Report
> Merging [#309](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=desc) into [master](https://codecov.io/gh/dusk-network/dusk-blockchain/commit/d68b0e10f2361acc603cf298ba2762a87fd80441&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #309   +/-   ##
=======================================
  Coverage   53.41%   53.41%           
=======================================
  Files         128      128           
  Lines        7185     7185           
=======================================
  Hits         3838     3838           
  Misses       2831     2831           
  Partials      516      516           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=footer). Last update [d68b0e1...5b88327](https://codecov.io/gh/dusk-network/dusk-blockchain/pull/309?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: pkg/p2p/peer/processing/chainsync/sync.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (s *ChainSynchronizer) Synchronize(blkBuf *bytes.Buffer, peerInfo string) e
 			return err
 		}
```
Comment (ggoranov) [inline pkg/p2p/peer/processing/chainsync/sync.go]: Is reader from `r := bufio.NewReader(blkBuf)` redundant? 

```go

// Write bufio.Reader into a bytes.Buffer and unmarshal it so we can send it over the event bus.
		buf := new(bytes.Buffer)
		if _, err := buf.ReadFrom(r); err != nil {
			return err
		}

		if err := message.UnmarshalBlock(buf, blk); err != nil {
			return err
		}
```

Instead, we can use ` message.UnmarshalBlock(blkBuf, blk);` or I'm missing something.

Comment (julesdesmit) [reply pkg/p2p/peer/processing/chainsync/sync.go]: It's not - we use it to peek the block height at the start of the function. Helps us determine whether we should start syncing, forward the block, or discard it.

Comment (ggoranov) [reply pkg/p2p/peer/processing/chainsync/sync.go]: That's correct. Forgot this. Thanks!

**Pull request merged**